### PR TITLE
feat(vats): upgradeable mintHolder contract

### PR DESCRIPTION
**Pull request by dckc**
refs: #6553

## Description

Enhance `mintHolder` to not only `makeDurableIssuerKit` but also re-connect it to `instanceBaggage` on upgrade.

### Documentation Considerations

makes for sort of a "hello world" upgrade example, maybe?

### Testing Considerations

SwingSet test of null upgrade based on [test-coveredCall-service-upgrade.js](https://github.com/Agoric/agoric-sdk/blob/57d12d66600108e57baa3cb915799c682be6446d/packages/zoe/test/swingsetTests/upgradeCoveredCall/test-coveredCall-service-upgrade.js); perhaps not the latest-and-greatest upgrade test harness, but pretty straightforward in this case.

### CODE REVIEW

File: packages/vats/src/mintHolder.js [hunk 1]
```diff
@@ -17,19 +17,25 @@ import { makeDurableIssuerKit } from '@agoric/ertp';
  * @param {unknown} _privateArgs
```
Comment (turadg) [inline packages/vats/src/mintHolder.js]: not your change but might be worth tightening this up
```suggestion
 * @param {undefined} _privateArgs
```

File: packages/vats/src/mintHolder.js [hunk 2]
```diff
@@ -17,19 +17,25 @@ import { makeDurableIssuerKit } from '@agoric/ertp';
...
-
-  const { mint, issuer } = makeDurableIssuerKit(
-    instanceBaggage,
-    keyword,
-    assetKind,
-    displayInfo,
-  );
+export const prepare = (zcf, _privateArgs, instanceBaggage) => {
+  const provide = () => {
+    if (!instanceBaggage.has('name')) {
```
Comment (turadg) [inline packages/vats/src/mintHolder.js]: this `name` value looks a little magic. What code defined this string? Can we have a shared constant?

File: packages/vats/src/mintHolder.js [hunk 3]
```diff
@@ -17,19 +17,25 @@ import { makeDurableIssuerKit } from '@agoric/ertp';
...
-  const { keyword, assetKind, displayInfo } = zcf.getTerms();
-
-  const { mint, issuer } = makeDurableIssuerKit(
-    instanceBaggage,
-    keyword,
-    assetKind,
-    displayInfo,
-  );
+export const prepare = (zcf, _privateArgs, instanceBaggage) => {
+  const provide = () => {
```
Comment (turadg) [inline packages/vats/src/mintHolder.js]: this contract is simple but for consistency with providers in more complex ones, consider defining at module scope,
```
const provideDurableIssuerKit = (baggage, zcf) => {
```

File: packages/vats/test/upgrading/bootstrap.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    // null upgrade
+    upgradeV1: async () => {
+      const [{ bundleId }, { facets, brand: expectedBrand }] =
+        await Promise.all([bootKit.promise, v1Kit.promise]);
+
+      const upgradeResult = await E(facets.adminFacet).upgradeContract(
+        bundleId,
+      );
+      assert.equal(upgradeResult.incarnationNumber, 2); // move to test assertion?
```
Comment (turadg) [inline packages/vats/test/upgrading/bootstrap.js]: makes sense here imo. it's `upgradeV1` checking itself. to use `t` would require letting the test peek in.

File: packages/vats/test/upgrading/test-upgrade-contracts.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+/** @type {import('ava').TestFn<Awaited<ReturnType<typeof makeTestContext>>>} */
+const test = anyTest;
+
+/**
+ * WARNING: uses ambient authority
+ * of import.meta.url
+ *
+ * We aim to use ambient authority only in test.before();
+ * splitting out makeTestContext() lets us share type info.
```
Comment (turadg) [inline packages/vats/test/upgrading/test-upgrade-contracts.js]: it's only used once here so "share" is a little confusing. 
```suggestion
 * splitting out makeTestContext() lets us type t.context
```

**Review by Chris-Hibbert — APPROVED** [review]
lgtm

File: packages/vats/test/upgrading/vat-ertp-service.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { Far } from '@endo/marshal';
+import { AmountMath } from '@agoric/ertp';
+import { prepareErtpService } from '@agoric/ertp/test/swingsetTests/ertpService/vat-ertp-service.js';
+
+export const buildRootObject = async (vatPowers, _vatParams, baggage) => {
```
Comment (turadg) [inline packages/vats/test/upgrading/vat-ertp-service.js]: stray inclusion?

**Review by turadg — APPROVED** [review]

File: packages/vats/test/upgrading/bootstrap.js (hunk 4 above)
Comment (dckc) [reply packages/vats/test/upgrading/bootstrap.js]: I thought about returning the `incarnationNumber` and checking in the test code; but returning stuff across the kernel boundary might be tedious.

File: packages/vats/test/upgrading/test-upgrade-contracts.js (hunk 5 above)
Comment (dckc) [reply packages/vats/test/upgrading/test-upgrade-contracts.js]: The type is used in `test('...', async t => {...})`; i.e. it's shared between `test.before()` and `test()`. But "lets us type..." is fine by me.

File: packages/vats/src/mintHolder.js (hunk 2 above)
Comment (Chris-Hibbert) [reply packages/vats/src/mintHolder.js]: I think the way I would do this is to add a method to issuerKit: 
```
  const hasIssuer = baggage => {
    return issuerBaggage.has('name') &&
       issuerBaggage.has('assetKind') &&
       issuerBaggage.has('displayInfo') &&
       issuerBaggage.has('elementShape');
   };
```

File: packages/vats/test/upgrading/vat-ertp-service.js (hunk 6 above)
Comment (dckc) [reply packages/vats/test/upgrading/vat-ertp-service.js]: yes; since removed

File: packages/vats/src/mintHolder.js (hunk 2 above)
Comment (turadg) [reply packages/vats/src/mintHolder.js]: what that's doing is much clearer to me. no need for constants in that case

Comment (dckc) [reply packages/vats/src/mintHolder.js]: I defined it that way, but haven't moved it to `@agoric/ertp` yet. Would you deep import it from `packages/ERTP/src/issuerKit.js`, @Chris-Hibbert ? Or really make it part of the public ERTP API?

Comment (Chris-Hibbert) [reply packages/vats/src/mintHolder.js]: I think a deep import is fine in this case. Whoever is calling it is sharing implementation details with ERTP.

Comment (dckc) [reply packages/vats/src/mintHolder.js]: turns out there's an `export * from './issuerKit.js';`

**Review by turadg — APPROVED** [review]
:niice:

**Pull request merged**